### DISCUSSION TITLE
I use websocket to  receive binary inputstream  in netty,  I want to forward this binary stream directly to other places in a zero-copy manner. What should I do

### DISCUSSION THREAD

**Discussion by yangh0597** [netty/netty#13456] [General]
I use websocket to  receive binary inputstream  in netty,  I want to forward this binary stream directly to other places in a zero-copy manner. What should I do. For example, write directly to a file 。Use The method ofFileChannel#transferTo()

![image](https://github.com/netty/netty/assets/86940083/b213cc87-f3f1-466c-94b5-73dc4950bae4)

**Comment by normanmaurer** [comment]
It depends a bit on where you want to transfer this too... For example if you want to send it to another `Channel` you can just do `channel.write(...)`.

**Reply by yangh0597** [reply]
thanks, Can i use ` channel.write(...)` write a DirectByteBuf ?  Now I use the bytes array to DirectByteBuf to read the data to the heap, and then write the data out immediately. I think this reads heap causing the code to slow down. Is there any plan to improve it?
![image](https://github.com/netty/netty/assets/86940083/5b5afc51-7890-4b7b-bc39-55622d3d6d49)

**Reply by chrisvest** [reply]
The `channel.write` method sends a message down the pipeline to the next handler, so what is accepted depends on the handlers. If the message reaches the end of the pipeline, then it goes to I/O and in that case, yes, it can take direct `ByteBuf` instances. In fact, direct buffers are preferred for I/O.

**Reply by yangh0597** [reply]
thanks
